src/ast/ast_build_filter_tree.c

```diff
@@ -232,17 +254,39 @@ FT_FilterNode *_FilterNode_FromAST(const cypher_astnode_t *expr) {
 	}
 }
 
-void _AST_ConvertFilters(const AST *ast, FT_FilterNode **root, const cypher_astnode_t *entity) {
+void _AST_ConvertGraphEntities(const AST *ast, FT_FilterNode **root,
+							   const cypher_astnode_t *entity) {
 	if(!entity) return;
 
-	cypher_astnode_type_t type = cypher_astnode_type(entity);
-
 	FT_FilterNode *node = NULL;
+	cypher_astnode_type_t type = cypher_astnode_type(entity);
 	// If the current entity is a node or edge pattern, capture its properties map (if any)
 	if(type == CYPHER_AST_NODE_PATTERN) {
 		node = _convertInlinedProperties(ast, entity, GETYPE_NODE);
 	} else if(type == CYPHER_AST_REL_PATTERN) {
 		node = _convertInlinedProperties(ast, entity, GETYPE_EDGE);
+	} else {
+		uint child_count = cypher_astnode_nchildren(entity);
+		for(uint i = 0; i < child_count; i++) {
+			const cypher_astnode_t *child = cypher_astnode_get_child(entity, i);
+			// Recursively continue searching
+			_AST_ConvertGraphEntities(ast, root, child);
```

Comment:
I prefer to avoid excessive recursive visits to the AST. This function accepts a pattern, so we can loop over each contained path (`cypher_ast_pattern_npaths()`) and `_convertInlinedProperties` for each element - nodes at even offsets, edges at odd.

Reply:
done